README.md

```diff
@@ -87,7 +87,19 @@ Unlike many authorization systems where a service controls access to resources i
 
 # 2. Terminology
 
-## 2.1 Resource
+## 2.1 Roles
+
+There are several roles that an agent MAY assume:
+
+| Name      | Description | 
+| --------- | ----------- |
+| Agent     | The general class of entities and principals that interact with a UCAN |
+| Principal | A DID listed in a UCAN's `iss` or `aud` field |
+| Delegator | The principal signing the current UCAN, and listed in the `iss` field |
+| Delegate  | The principal being delegated to in the current UCAN, listed in the `aud` field |
+| Validator | Any party that interprets a UCAN to determine that it is valid, and which capabilities it grants |
```

Comment:
The way i think about it is:
- `Agent` can be an Issuer or a Audience
- `Principal` generic denomination for DID
- `Issuer` is a principal with a private key
- `Audience` is just a Principal that we need a special word for

Reply:
Actually `Agent` and `Issuer` can be the same thing depends on the side.
Example:

From Agent1 side he is the Agent (Principal with private key) and Audience is just a DID (Agent2 DID).
From Agent2 side reverse.

So Agent can't really be Issuer or Audience, its more a question of locality (local vs remote).

So maybe this is enough:
- `Principal` is just a DID (may or may not have a keypair)
- `Agent` is a local `Principal` that can issue UCANs with a keypair to another `Principal`

Reply:
@hugomrdias I think thinking of `Issuer` as an entity with private key is misleading, because it depends on the perspective. On the sender side UCAN issuer is the entity with private key as it needs to sign that UCAN, however on the receiver side UCAN issuer is no longer entity with a private key or at least representation of it can not sign anything.

I am afraid `Audience` is the same you're on the receiver end audience will be an entity with signing capabilities, however on the sender side audience will be a role with verification capabilities.